code/datums/components/wire_panel.dm

```diff
@@ -0,0 +1,284 @@
+/**
+ * Wire Panel Component
+ *
+ * Addding this component to an object will add an interactable wire panel.
+ *
+ * Consumer is responsible for icon updates / functionality changes based on wire control.
+ */
+
+TYPEINFO(/datum/component/wirePanel)
+	initialization_args = list(
+		ARG_INFO("definition", DATA_INPUT_REF, "Datum holding our wire panel defintion"),
+	)
+/datum/component/wirePanel
+	/// Reference to a wire defintion datum, passed in Initialize
+	var/datum/wirePanel/panelDefintion/panel_def
+	/// panel cover status (default closed)
+	var/cover_status = WPANEL_COVER_CLOSED
+	/// wire-indexed list of booleans
+	var/list/cut_wires = list()
+	/// bitmask of which wire controls are active (default all active)
+	var/active_wire_controls = ~0
```

Comment:
is there a reason for the mix of bitmasks and lists here?

Reply:
the list allows for the UI to easily display the wires in the same order across all of the same machine type. 

Using a bitmask for the machine controls means we can glean many on/off values off the response of a single signal.

Further, having a logical separation (i.e. list vs. bitmask) makes it clear that the order of the wires has no influence on the order of the active controls.